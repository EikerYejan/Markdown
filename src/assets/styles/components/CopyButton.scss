@import "../common/common";

.copy-button {
  @include flexCenter;
  padding: 6 * $px;
  position: absolute;
  top: 0;
  right: 0;
  width: 40 * $px;
  height: 40 * $px;
  background: none;
  border: none;
  border-color: $borders;
  border-width: 1 * $px;
  border-bottom-style: solid;
  border-left-style: solid;
  border-right-style: solid;
  cursor: pointer;
  background: $black;

  img {
    width: 100%;
    height: 100%;
    filter: invert(1);
  }

  &.copy-html {
    border-right-style: unset;
  }
}

@media screen and (max-width: 768 * $px) {
  .copy-button {
    border-right-style: unset;
  }
}
