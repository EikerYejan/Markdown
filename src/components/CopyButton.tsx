import React from "react"
import copy from "copy-to-clipboard"
import toast from "cogo-toast"
import clipboard from "../assets/images/clipboard.svg"
import "../assets/styles/components/CopyButton.scss"

type HandleCopy = (e: React.SyntheticEvent<HTMLButtonElement>) => void

type Props = {
  toCopy: string
  className?: string
}

const CopyButton: React.FC<Props> = ({ toCopy, className }) => {
  /**
   * Copy text to clipboard
   */
  const handleCopy: HandleCopy = (e) => {
    /* Prevent default */
    e.preventDefault()

    /* Copy to clipboard */
    copy(toCopy)

    /* Show toast */
    toast.success("Copied to clipboard", {
      position: "top-right",
      hideAfter: 2,
    })
  }

  return (
    <button
      className={`copy-button has-tooltip ${className ?? ""}`}
      title="Copy to clipboard"
      onClick={handleCopy}
      type="button"
    >
      <img src={clipboard} alt="copy-markdown" />
    </button>
  )
}

export default CopyButton
