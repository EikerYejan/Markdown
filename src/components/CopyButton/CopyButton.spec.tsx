import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import CopyButton from '.'

jest.mock('copy-to-clipboard')
jest.mock('cogo-toast')

describe('<CopyButton', () => {
  it('Should render', () => {
    const { baseElement } = render(<CopyButton toCopy="TEXT" />)
    const button = baseElement.querySelector('button')

    if (!button) throw new Error('Button not found')

    userEvent.click(button)
    expect(baseElement).toBeInTheDocument()
    expect(baseElement).toMatchSnapshot()
  })
})
