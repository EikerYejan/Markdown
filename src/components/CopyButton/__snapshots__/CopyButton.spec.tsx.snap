// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CopyButton Should render 1`] = `
<body>
  <div>
    <button
      class="copy-button has-tooltip "
      title="Copy to clipboard"
      type="button"
    >
      <img
        alt="copy-markdown"
        src="clipboard.svg"
      />
    </button>
  </div>
</body>
`;
